### PR TITLE
feat: Add support for code_scanning

### PR DESCRIPTION
**Pull request by ihor-hrytskiv**
Suggestion to add support for code_scanning in Github API.

[Ruleset API docs](https://docs.github.com/en/rest/orgs/rules?apiVersion=2022-11-28#create-an-organization-repository-ruleset)

### CODE REVIEW

File: github/repos_rules.go [hunk 1]
```diff
@@ -130,6 +130,18 @@ type RequiredWorkflowsRuleParameters struct {
...
+// RuleRequiredCodeScanningTools represents the RequiredCodeScanningTools for the RequiredCodeScanningParameters object.
+type RuleRequiredCodeScanningTools struct {
+	AlertsThreshold         string `json:"alerts_threshold"`
+	SecurityAlertsThreshold string `json:"security_alerts_threshold"`
+	Tool                    string `json:"tool"`
+}
+
+// RequiredCodeScanningRuleParameters represents the code_scanning rule parameters.
+type RequiredCodeScanningRuleParameters struct {
+	RequiredCodeScanningTools []RuleRequiredCodeScanningTools `json:"code_scanning_tools"`
```
Comment (gmlewis) [inline github/repos_rules.go:142]:
```suggestion
	RequiredCodeScanningTools []*RuleRequiredCodeScanningTools `json:"code_scanning_tools"`
```

**Review by gmlewis — APPROVED** [review]
Thank you, @ihor-hrytskiv !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
@ihor-hrytskiv - can you please resolve the conflicts in this PR and push (not force-push) the updates when you get a chance?  Hopefully we will then get a second LGTM+Approval and can proceed to merge.  Thank you!

File: github/repos_rules_test.go [hunk 2]
```diff
@@ -305,6 +305,34 @@ func TestRepositoryRule_UnmarshalJSON(t *testing.T) {
 				},
 			}),
 		},
+		"Invalid Required workflows params": {
```
Comment (air-hand) [inline github/repos_rules_test.go:308]: 👍

**Review by air-hand — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @air-hand !
Updating branch...

**Pull request merged**